core/processor/src/main/java/io/quarkus/annotation/processor/generate_doc/ConfigDoItemFinder.java

```diff
@@ -162,19 +163,19 @@ ScannedConfigDocsItemHolder findInMemoryConfigurationItems() {
 
             if (isConfigGroup) {
                 List<ConfigDocItem> groupConfigItems = recordConfigItemsFromConfigGroup(configPhase, name, configGroup,
-                        configSection, withinAMap, sectionLevel);
+                        configSection, withinAMap, sectionLevel, optional);
```

Comment:
I don't think it is correct to make all members of an optional group be optional.  In fact an optional config group can have required members.  In this case, the required members are all required if any member of the group was given.

Reply:
That's true, thanks for clarifying the behaviours. 

We can change this to maybe add a section indicating that it is optional. The members will be left untouched. WDYT?

Reply:
Sounds good.